src/scripts/cronjobs/firstDataBrokerRemovalFixed.tsx

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { OnerepScanResultRow, SubscriberRow } from "knex/types/tables";
+import {
+  getPotentialSubscribersWaitingForFirstDataBrokerRemovalFixedEmail,
+  markFirstDataBrokerRemovalFixedEmailAsJustSent,
+} from "../../db/tables/subscribers";
+import { initEmail, sendEmail } from "../../utils/email";
+import { renderEmail } from "../../emails/renderEmail";
+import { FirstDataBrokerRemovalFixed } from "../../emails/templates/firstDataBrokerRemovalFixed/FirstDataBrokerRemovalFixed";
+import { getEmailL10n } from "../../app/functions/l10n/cronjobs";
+import { sanitizeSubscriberRow } from "../../app/functions/server/sanitize";
+import { refreshStoredScanResults } from "../../app/functions/server/refreshStoredScanResults";
+import { getLatestOnerepScanResults } from "../../db/tables/onerep_scans";
+
+type SubscriberFirstRemovedScanResult = {
+  subscriber: SubscriberRow;
+  firstRemovedScanResult: OnerepScanResultRow;
+};
+
+function isFulfilledResult(
+  result: PromiseSettledResult<SubscriberFirstRemovedScanResult | undefined>,
+): result is PromiseFulfilledResult<SubscriberFirstRemovedScanResult> {
+  return typeof result !== "undefined" && result.status === "fulfilled";
+}
+
+void run();
+
+async function run() {
+  const batchSize = Number.parseInt(
+    process.env.FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE ?? "10",
+    10,
+  );
+  if (Number.isNaN(batchSize)) {
+    throw new Error(
+      `Could not send first data broker removal fixed emails, because the env var FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE has a non-numeric value: [${process.env.FIRST_DATA_BROKER_REMOVAL_FIXED_EMAIL_BATCH_SIZE}].`,
+    );
+  }
+  const potentialSubscribersToEmail =
+    await getPotentialSubscribersWaitingForFirstDataBrokerRemovalFixedEmail();
+
+  const subscribersToEmailWithData = (
+    await Promise.allSettled(
+      potentialSubscribersToEmail.map(async (subscriber) => {
+        // OneRep suggested not relying on webhooks, but instead to fetch the latest
+        // data from their API. Thus, let's refresh the data in our DB in real-time:
+        if (subscriber.onerep_profile_id !== null) {
+          await refreshStoredScanResults(subscriber.onerep_profile_id);
+        }
+        const latestScan = await getLatestOnerepScanResults(
+          subscriber.onerep_profile_id,
+        );
+
+        let firstRemovedScanResult = null;
+        for (const scanResult of latestScan.results) {
```

Comment:
This logic can probably be simplified with `.filter.reduce` chain
```
const firstRemovedScanResult = latestScan.results
        .filter(scanResult => !scanResult.manually_resolved && scanResult.status === "removed")
        .reduce((first, scanResult) => {
        ...
```

Reply:
This was my first thought as well. I guess it does not matter too much, because the results likely won’t be a huge list, but with the current implementation, we only have to go over them once.